otelcol/collector.go

```diff
@@ -242,23 +235,50 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	if err = col.service.Start(ctx); err != nil {
 		return multierr.Combine(err, col.service.Shutdown(ctx))
 	}
-	col.setCollectorState(StateRunning)
 
+	col.setCollectorState(StateRunning)
 	return nil
 }
 
+func (col *Collector) loadConfiguration(ctx context.Context) (*Config, Factories, error) {
+	factories, err := col.set.Factories()
+	if err != nil {
+		return nil, Factories{}, fmt.Errorf("failed to initialize factories: %w", err)
+	}
+
+	cfg, err := col.configProvider.Get(ctx, factories)
+	if err != nil {
+		return nil, Factories{}, fmt.Errorf("failed to get config: %w", err)
+	}
+
+	if err := xconfmap.Validate(cfg); err != nil {
+		return nil, Factories{}, fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	return cfg, factories, nil
+}
+
 func (col *Collector) reloadConfiguration(ctx context.Context) error {
-	col.service.Logger().Warn("Config updated, restart service")
-	col.setCollectorState(StateClosing)
+	logger := col.service.Logger()
+	logger.Info("Reloading collector configuration...")
```

Comment:
Should this and the corresponding "reloaded configuration" logs be debug logs? We already print that we're handling the signal and the Collector will print that the service is starting/stopping.